apps/deploy-web/src/pages/billing/index.tsx

```diff
@@ -0,0 +1,11 @@
+import { BillingPage } from "@src/components/billing-usage/BillingPage";
+import { RegisteredUsersOnly } from "@src/hoc/registered-users-only/registered-users-only.hoc";
+import { defineServerSideProps } from "@src/lib/nextjs/defineServerSideProps/defineServerSideProps";
+import { isAuthenticated, isFeatureEnabled } from "@src/lib/nextjs/pageGuards/pageGuards";
+
+export default RegisteredUsersOnly(BillingPage);
+
+export const getServerSideProps = defineServerSideProps({
+  route: "/billing",
+  if: async ctx => (await isAuthenticated(ctx)) && (await isFeatureEnabled("billing_usage", ctx))
```

Comment:
we have a HOC for this called `withCustomPageAuthRequired`, perhaps we can modify it accept some feature list to check for as well to check for `isFeatureEnabled`

Reply:
imo this approach is more verbose an transparent. let's discuss pls

Reply:
Let's keep like this for now and we'll address this in another pr